src/main/java/org/folio/circulation/domain/MoveRequestProcessAdapter.java

```diff
@@ -36,7 +38,12 @@ public MoveRequestProcessAdapter(ItemRepository itemRepository, LoanRepository l
   }
 
   CompletableFuture<Result<RequestAndRelatedRecords>> getDestinationRequestQueue(
-      RequestAndRelatedRecords requestAndRelatedRecords) {
+    RequestAndRelatedRecords requestAndRelatedRecords) {
+    Request request = requestAndRelatedRecords.getRequest();
+    if (request.isTlrFeatureEnabled()) {
```

Comment:
It would be great to incapsulate this choice we make based on TLR-settings inside `RequestQueueRepository`. It already has method `getQueue(LoanAndRelatedRecords records)`. Can we make it more generic? Something like:

```
getQueue(TlrSettingsConfiguration tlrSettings, Item item) {
  return tlrSettings != null && tlrSettings.isTitleLevelRequestsFeatureEnabled()
      ? getByInstanceId(item.getInstanceId())
      : getByItemId(item).getItemId());
}
```
Then we can overload it for purposes of specific resources:
```
getQueue(RequestAndRelatedRecords requestAndRelatedRecords) {
  Request request = requestAndRelatedRecords.getRequest();
  return getQueue(request.getTlrSettingsConfiguration(), request.getItem());
}
```
The idea is to have just **one** place where we check TLR settings in order to determine how the queue should be built.

Reply:
Great idea, but we won't be able to use it in case of creating a new request, because in the case of Hold TLR we don't have an item. I've used this approach for checkout and moving request cases.